pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -269,15 +276,26 @@ func (e *EndpointController) processNextWorkItem() bool {
 	defer e.queue.Done(eKey)
 
 	err := e.syncService(eKey.(string))
+	e.handleErr(err, eKey)
+
+	return true
+}
+
+func (e *EndpointController) handleErr(err error, key interface{}) {
 	if err == nil {
-		e.queue.Forget(eKey)
-		return true
+		e.queue.Forget(key)
+		return
 	}
 
-	utilruntime.HandleError(fmt.Errorf("Sync %v failed with %v", eKey, err))
-	e.queue.AddRateLimited(eKey)
+	if e.queue.NumRequeues(key) < maxRetries {
+		glog.V(2).Infof("Error syncing endpoints for service %v: %v", key, err)
```

Comment:
- [ ] Here you format the "key" value with `%v`, but at line 297 below you use `%q`. Be consistent.  
  I prefer `%q` for string values provided by users.